src/Form/ItemCheckbox.js

```diff
@@ -0,0 +1,116 @@
+import classNames from 'classnames';
+import React from 'react';
+
+import BindMixin from '../Mixin/BindMixin';
+import Util from '../Util/Util';
+
+export default class ItemCheckbox extends Util.mixin(BindMixin) {
+  get methodsToBind() {
+    return ['handleChange'];
+  }
+
+  componentDidMount() {
+    let {props, refs} = this;
+    let checkbox = React.findDOMNode(refs.checkbox);
+
+    if (props.checked != null) {
+      checkbox.checked = props.checked;
+    }
+
+    let indeterminate = props.indeterminate;
+    if (indeterminate != null) {
+      checkbox.indeterminate = indeterminate;
+    }
+  }
+
+  componentDidUpdate() {
+    let {props, refs} = this;
+    let checkbox = React.findDOMNode(refs.checkbox);
+
+    if (props.checked != null) {
```

Comment:
we should pull this out into a method called `updateCheckbox` or something. it's the same code twice